### PR TITLE
Adds documentation for CSR supporting all issuers, and the CTL command

### PR DESCRIPTION
**Pull request by JoshVanL**
Signed-off-by: joshvanl <vleeuwenjoshua@gmail.com>

This PR documents the support for all CSR signer types, including their used annotations.

Also documents the `cert-manager x create csr` command.

---

This PR has been updated to remove the `experimental.cert-manager.io/ca` annotation, which has been removed from cert-manager.

```release-note
NONE
```

### CODE REVIEW

File: content/en/docs/usage/kube-csr.md
Comment (irbekrm) [inline content/en/docs/usage/kube-csr.md:128]: Nit: could we state somewhere that in this doc 'signer' means the same as 'issuer/clusterisuer'? It is somehwat inferrable, but I imagine may still be confusing.

File: content/en/docs/usage/kubectl-plugin.md
Comment (irbekrm) [inline content/en/docs/usage/kubectl-plugin.md:166]: Nit- so that's it's obvious this doesn't talk about plugin as a whole
```suggestion
`kubectl cert-manager x` has experimental sub-commands for operations which are
```

**Review by irbekrm — APPROVED** [review]
Looks good and clear to me.

Adding an lgtm with a hold if you want to address the two optional nits I added

/lgtm
/hold

**Comment by JoshVanL** [comment]
Thanks @irbekrm, can you take another look?

**Comment by irbekrm** [comment]
Thanks for adding the changes, I've read through the doc again and it looks good

/lgtm

**Pull request merged**
